src/libsyncengine/update_detection/update_detector/updatetreeworker.h

```diff
@@ -134,7 +134,7 @@ class UpdateTreeWorker : public ISyncWorker {
          */
         ExitCode handleCreateOperationsWithSamePath();
 
-        std::shared_ptr<Node> getOrCreateNodeFromPath(const SyncPath &path);
+        std::shared_ptr<Node> getOrCreateNodeFromPath(const SyncPath &path, const bool isDeleted);
```

Comment:
It doesn't seem to be easy, but here is a wish: would it be possible to find a suggestive name for 

```std::shared_ptr<Node> aCoolAndMeaningfulName(const SyncPath &path) { return getOrCreateNodeFromPath(path, true);}``` and ```std::shared_ptr<Node> anotherCoolAndMeaningfulName(const SyncPath &path) { return getOrCreateNodeFromPath(path, false);}```